### DISCUSSION TITLE
Post inpainting / "upscaled inswapper"

### DISCUSSION THREAD

**Discussion by a3nima** [glucauze/sd-webui-faceswaplab#6] [Q&A] [answered]
1. In img2img I wanted to swap one face. This worked as intended. Then I've tried to use the inpainting area, but the extension started working on all faces present. How can it be done to let it only on the selected face in the image ?
2. In the documentation I see and read about upscaled inswapper, is this obsolete or is this somehow the Post inpainting?

Thanks a lot and thanks for the development of Faceswaplab!

**Comment by glucauze** [comment]
1. I'm missing some information. Post-processing does apply to all images (whether exchanged or not). The face index takes all faces into account (even if they're not masked, you have to be careful). But faces outside the mask will not be exchanged.

2. The upscaled inswapper only concerns swapped faces. And it can be activated from the SD parameters. It's not the same thing at all. See the Processing Order section of the doc.

**Reply by glucauze** [reply, marked as answer]
see here for reference : https://github.com/glucauze/sd-webui-faceswaplab/issues/7
